server/src/main/java/org/opensearch/action/admin/indices/shrink/TransportResizeAction.java

```diff
@@ -176,12 +180,16 @@ static CreateIndexClusterStateUpdateRequest prepareCreateIndexRequest(
         targetIndexSettingsBuilder.remove(IndexMetadata.SETTING_HISTORY_UUID);
         final Settings targetIndexSettings = targetIndexSettingsBuilder.build();
         final int numShards;
+        ByteSizeValue maxShardSize = resizeRequest.getMaxShardSize();
         if (IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.exists(targetIndexSettings)) {
             numShards = IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.get(targetIndexSettings);
+            if (resizeRequest.getResizeType() == ResizeType.SHRINK && maxShardSize != null) {
```

Comment:
Why only for `ResizeType.SHRINK`? Shouldn't it be an invalid combination for other resize types?

Reply:
This parameter `max_shard_size` is applicable to `Shrink` only,  because when users want to shrink an index, their intention is to reduce the number of shards, so we can use this parameter to find the lowest factor of the source index's shards count while each shard's storage do not exceed the give value, we should reduce the number of shards as much as possible.  Other resize types include `Clone` and `Split`, it's not applicable for `Clone` because we cannot change the number_of_shards of the source index, and for `Split`, I think it's also not applicable, because users want to increase the number of shards, if `max_shard_size` is set, many candidate numbers can be chosen as the shards count in the new split index, and we don't know whether the lowest multiple of the source index's shards is optimum or not, for example, if the source index has  5 primary shards with totaling 300gb, and `max_shard_size` is 30gb, all the number 10, 20, 30... satisfy the requirements, but we cannot select 10 simply because users may want to increase the shards count of the source index by multiple times but not doubly.

Reply:
@gaobinlong that's basically my question: if `max_shard_size` is not applicable but provided, shouldn't we throw `IllegalArgumentException` ?

Reply:
Yeah, we should throw `IllegalArgumentException` for other resize types, I will do it.

Reply:
@gaobinlong I think this one is still not addressed?